### PR TITLE
LENS-33 - Add ENS resolution to Linea ENS

### PR DESCRIPTION
**Pull request by igorms-cons**
## Related ticket

Fixes [LENS-33](https://consensyssoftware.atlassian.net/browse/LENS-33)

## Type of change

- [ ] Bug fix (non-breaking change, which fixes an issue)
- [x] New feature (non-breaking change, which adds functionality)
- [ ] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

[LENS-33]: https://consensyssoftware.atlassian.net/browse/LENS-33?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by igorms-cons** [comment]
Screenshot 2023-05-31 at 15 29 47

File: apps/web/src/components/Profile/Details.tsx [hunk 1]
```diff
@@ -24,12 +30,14 @@ import Link from 'next/link';
...
 import type { Dispatch, FC, ReactElement } from 'react';
-import { useState } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { useAppStore } from 'src/store/app';
 import { useMessageStore } from 'src/store/message';
 import { FollowSource } from 'src/tracking';
 import { Button, Image, Modal, Tooltip } from 'ui';
+import { useAccount, useContractRead } from 'wagmi';
 
+import { ENS_FRONT_DEV_LINEA_URL } from '../../../packages/data/constants';
```
Comment (alainncls) [inline apps/web/src/components/Profile/Details.tsx:40]: This should be imported from `data/constants` (cf. line 20)

File: apps/web/src/components/Profile/Details.tsx [hunk 2]
```diff
@@ -41,14 +49,84 @@ interface DetailsProps {
   setFollowing: Dispatch<boolean>;
 }
 
+interface Domain {
```
Comment (alainncls) [inline apps/web/src/components/Profile/Details.tsx:52]: Let's move it to `apps/web/src/types/index.d.ts`

File: apps/web/src/components/Profile/Details.tsx [hunk 3]
```diff
@@ -41,14 +49,84 @@ interface DetailsProps {
...
+  tokenId: string;
+  name: string | null;
```
Comment (alainncls) [inline apps/web/src/components/Profile/Details.tsx:54]: Maybe `name?:string`?

File: apps/web/src/components/Profile/Details.tsx [hunk 4]
```diff
@@ -41,14 +49,84 @@ interface DetailsProps {
...
+    address: LINEA_RESOLVER,
+    abi: LINEA_RESOLVER_ABI,
+    functionName: 'balanceOf',
+    args: [String(address)]
+  });
+
+  const balance = parseInt(balanceData as string);
+
+  if (isBalanceError) {
+    console.log('Linea resolver balanceOf error', balanceError?.message);
```
Comment (alainncls) [inline apps/web/src/components/Profile/Details.tsx:84]: To remove

File: apps/web/src/components/Profile/Details.tsx [hunk 5]
```diff
@@ -41,14 +49,84 @@ interface DetailsProps {
...
+          address: String(address),
+          registered: true,
+          owned: true
+        };
+        setDomain(domainData);
+      } catch (error: any) {
+        console.log('getTokenMetadata error', error?.message);
+      }
+    } else {
+      console.log('tokenOfOwnerByIndex error', TokenError);
```
Comment (alainncls) [inline apps/web/src/components/Profile/Details.tsx:120]: To remove

**Review by alainncls — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Profile/Details.tsx (hunk 3 above)
Comment (igorms-cons) [reply apps/web/src/components/Profile/Details.tsx:54]: null is not relevant

**Review by alainncls — CHANGES_REQUESTED** [review]

File: apps/web/src/components/Profile/Details.tsx [hunk 6]
```diff
@@ -23,32 +29,67 @@ import isVerified from 'lib/isVerified';
...
+    return await res.json();
+  };
+
+  const handleLineaDomain = useCallback(async () => {
+    if (!isTokenError) {
+      try {
+        const metadata = await getTokenMetadata(String(tokenId));
+        setDomain((metadata as any).name);
+      } catch (error: any) {
+        console.log('getTokenMetadata error', error?.message);
```
Comment (Akaryatrh) [inline apps/web/src/components/Profile/Details.tsx:82]: Do we want to keep this? if yes, maybe we could use `console.error` instead?

File: apps/web/src/components/Profile/Details.tsx [hunk 7]
```diff
@@ -23,32 +29,67 @@ import isVerified from 'lib/isVerified';
...
+    args: [String(address), 0]
+  });
+
+  const getTokenMetadata = async (tokenId: string) => {
+    const res = await fetch(`${ENS_FRONT_DEV_LINEA_URL}/${tokenId}`);
+    return await res.json();
+  };
+
+  const handleLineaDomain = useCallback(async () => {
+    if (!isTokenError) {
```
Comment (Akaryatrh) [inline apps/web/src/components/Profile/Details.tsx:77]: Can we move this condition to line 88?

File: packages/data/constants.ts [hunk 8]
```diff
@@ -179,3 +466,5 @@ export const S3_BUCKET = {
 };
 
 export const DISCORD_URL = 'https://discord.com/invite/9QwXqsyAps';
+
+export const ENS_DOMAIN_URL = 'https://app.ens.domains';
```
Comment (Akaryatrh) [inline packages/data/constants.ts:470]: Looks like https://app.ens.domains/ doesn't support Linea. Should we switch to https://ensfront.dev.linea.build/ specifically for Linea ENS domains ? cc @alainncls @mlallai

**Review by Akaryatrh — APPROVED** [review]

File: packages/data/constants.ts (hunk 8 above)
Comment (alainncls) [reply packages/data/constants.ts:470]: Yes, you have to switch to Ethereum Goerli to see your domain...  
But Linea ENS doesn't have an equivalent screen (I checked with the team), so we can't redirect there. Let's leave it like that, at least for the moment

**Pull request merged**